pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go

```diff
@@ -29,35 +29,28 @@ type CloudStackDatacenterConfigSpec struct {
 	// Domain contains a grouping of accounts. Domains usually contain multiple accounts that have some logical relationship to each other and a set of delegated administrators with some authority over the domain and its subdomains
 	Domain string `json:"domain"`
 	// Zones is a list of one or more zones that are managed by a single CloudStack management endpoint.
-	Zones []CloudStackZoneRef `json:"zones"`
+	Zones []CloudStackZone `json:"zones"`
 	// Account typically represents a customer of the service provider or a department in a large organization. Multiple users can exist in an account, and all CloudStack resources belong to an account. Accounts have users and users have credentials to operate on resources within that account. If an account name is provided, a domain name must also be provided.
 	Account string `json:"account,omitempty"`
+	// CloudStack Management API endpoint's IP. It is added to VM's noproxy list
+	ManagementApiEndpoint string `json:"managementApiEndpoint"`
 }
 
 type CloudStackResourceIdentifier struct {
-	Id   string `json:"id"`
-	Name string `json:"name"`
+	// +optional Id Unique CloudStack ID of the resource in the CLoudStack enironment. Mutually exclusive with Name
+	Id string `json:"id,omitempty"`
+	// +optional Name CloudStack Name of the resource in the CloudStack environment. Mutually exclusive with Id
+	Name string `json:"name,omitempty"`
```

Comment:
I think the comments are typically done above the `+` lines.

```suggestion
	// Id of a resource in the CloudStack environment. Mutually exclusive with Name
	// +optional
	Id string `json:"id,omitempty"`
	// Name of a resource in the CloudStack environment. Mutually exclusive with Id
	// +optional
	Name string `json:"name,omitempty"`
```